src/common/arrow/arrow_row_batch.cpp

```diff
@@ -755,8 +762,8 @@ ArrowArray* ArrowRowBatch::convertStructVectorToArray(ArrowVector& vector,
     result->children = vector.childPointers.data();
     result->n_children = (std::int64_t)StructType::getNumFields(type);
     for (auto i = 0u; i < StructType::getNumFields(type); i++) {
-        auto childType = StructType::getFieldTypes(type)[i];
-        vector.childPointers[i] = convertVectorToArray(*vector.childData[i], childType);
+        auto childType = StructType::getField(type, i).getType().copy();
```

Comment:
this should be a const& as I understand?

Reply:
Not if you immediately call copy